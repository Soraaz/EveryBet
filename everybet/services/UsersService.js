"use strict";

const HTTPClient = require("./HTTPClient");
const Common = require('./Common');

class UsersService {

    // POST

    static createNewUser(login, name, email, password, coins) {
        let body = {
            login: login,
            name: name,
            email : email,
            password: password,
            coins: coins
        };
        return HTTPClient.sendPost("users", body);
    }

    static addUserResponse(userId, betId, answerId, coins) {
        let body = {
            userId: userId,
            betId: betId,
            answerId: answerId,
            coins: coins
        };
        return HTTPClient.sendPost("users/answer", body);
    }

    static async dailyRewardByUserId(userId, coins) {
        await this.addCoinsByUserId(userId, coins);
        return await this.updateUserById(userId, "reward", "1");
    }

    static addImagetoUser(userId, img) {
        return HTTPClient.uploadImage("/images/upload/" + userId, img);
    }

    // PUT

    static updateUserById(id, columns, value) {
        let columnsMap = new Map();

        if (Array.isArray(columns) && Array.isArray(value))
            for (let i = 0; i < columns.length - 1; i++)
                columnsMap.set(columns[i], value[i]);
        else
            columnsMap.set(columns, value);
        return HTTPClient.sendPut("users/" + id, Common.parseColumns(columnsMap));
    }

    static updateUsersByFilters(filters, columns) {
        return HTTPClient.sendPut(Common.parseFilters("users", filters), Common.parseColumns(columns));
    }

    static updateAllUsers(columns, value) {
        if (arguments.length === 2) {
            let columnsMap = new Map();
            columnsMap.set(columns, value);
            columns = columnsMap;
        }
        return this.updateUsersByFilters([], columns);
    }

    static addCoinsByUserId(userId, coins) {
        let body = {
            "coins": coins
        };
        return HTTPClient.sendPut("users/" + userId + "/addCoins", body);
    }

    static addCoinsByFilters(filters, coins) {
        let body = {
            "coins": coins
        };
        return HTTPClient.sendPut(Common.parseFilters("users/addCoins", filters), body);
    }

    // DELETE

    static deleteUserById(id) {
        return HTTPClient.sendDelete("users/" + id);
    }

    static deleteUserByFilters(filters) {
        return HTTPClient.sendDelete(Common.parseFilters("users", filters));
    }

    static resetUserResponses(filters) {
        return HTTPClient.sendDelete(Common.parseFilters("users/answer", filters));
    }

    // GET

    static getUserById(id) {
        return HTTPClient.sendGet("users/" + id);
    }

    static getUsersByFilters(filters) {
        return HTTPClient.sendGet(Common.parseFilters("users", filters));
    }

    static getAllUsers() {
        return this.getUsersByFilters([]);
    }

    static getUsersByName(name) {
        let filters = new Map();
        filters.set("name", name);
        return this.getUsersByFilters(filters);
    }

    static getUsersByLogin(login) {
        let filters = new Map();
        filters.set("login", login);
        return this.getUsersByFilters(filters);
    }

    static getUserByEmail(email) {
        let filters = new Map();
        filters.set("email", email);
        return this.getUsersByFilters(filters);
    }
}

module.exports = UsersService;
