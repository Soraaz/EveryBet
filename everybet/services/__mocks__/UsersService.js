"use strict";

class UsersService {
    static addUserResponse = jest.fn().mockReturnValue(new Promise(async (resolve) => {
        resolve();
    }));

    static removeCoinsByUserId = jest.fn().mockReturnValue(new Promise(async (resolve) => {
        resolve();
    }));

    static addCoinsByUserId = jest.fn().mockReturnValue(new Promise(async (resolve) => {
        resolve();
    }));

    static getUserById = jest.fn().mockReturnValue(new Promise(async (resolve) => {
        let res = [{
            "additionalInformation": null,
            "address": null,
            "admin": 0,
            "avatar": "images/uploads/b77036a7cb9344dfc7b901e07097e036.png",
            "coins": 999999999,
            "country": null,
            "email": "a.a@a.fr",
            "expoPushToken": "",
            "id": 15,
            "login": "Altan",
            "name": "",
            "password": "43102984a36d9196dd2d92cf10de63cd44226bcc6f8e5135a9780a9c8303a44fb687ba6134b4d4b68259aac57d7ff0f2ab6ca877019b75752dda8cf03a1c18ac",
            "phone": null,
            "reward": 1,
            "rewardTier": 1,
            "rewardValue": 10,
            "tier": 1,
            "zipCode": null,
        }];
        resolve(res);
    }));

    static getUserByEmail = jest.fn().mockReturnValue(new Promise(async (resolve) => {
        let res = [{
            "additionalInformation": null,
            "address": null,
            "admin": 0,
            "avatar": null,
            "coins": 110,
            "country": null,
            "email": "antoine.dury@epitech.eu",
            "expoPushToken": "",
            "id": 3,
            "login": "Anto",
            "name": "Antoine",
            "password": "bf7420009c1334246948291c7df7925d25d8c1d845e822b54a418d28d6fc0a482c0229f06b177af26fb00fec0e876d13739e994af139f893cc61c7972929634c",
            "phone": null,
            "reward": 1,
            "rewardTier": 1,
            "rewardValue": 10,
            "tier": 1,
            "zipCode": null,
        }];
        resolve(res);
    }));

    static getAllUsers = jest.fn().mockReturnValue(new Promise(async (resolve) => {
        let res = [{
            "additionalInformation": null,
            "address": null,
            "admin": 0,
            "avatar": "images/uploads/ed8d8b46fbe0747ccf5735e0b88bfb2d.png",
            "coins": 11111,
            "country": null,
            "email": "Sora333@hotmail.fr",
            "expoPushToken": "",
            "id": 20,
            "login": "Sora2",
            "name": "",
            "password": "267f9f0b7fb7ad7209367f9b37db600c51b4729ca2eb842a7c125e107fd2531c36e19cbdc0cfee35d15ef927a1a222fab97742546cd973c69b145781036a080d",
            "phone": null,
            "reward": 1,
            "rewardTier": 1,
            "rewardValue": 10,
            "tier": 1,
            "zipCode": null,
        }, {
            "additionalInformation": null,
            "address": null,
            "admin": 0,
            "avatar": null,
            "coins": 10000,
            "country": null,
            "email": "Jaja@gmail.com",
            "expoPushToken": "",
            "id": 18,
            "login": "Jaja",
            "name": "Ja",
            "password": "bf7420009c1334246948291c7df7925d25d8c1d845e822b54a418d28d6fc0a482c0229f06b177af26fb00fec0e876d13739e994af139f893cc61c7972929634c",
            "phone": null,
            "reward": 1,
            "rewardTier": 1,
            "rewardValue": 10,
            "tier": 1,
            "zipCode": null,
        }, {
            "additionalInformation": null,
            "address": null,
            "admin": 0,
            "avatar": "images/uploads/456a8be835912a7f1e195607be360274.png",
            "coins": 509,
            "country": null,
            "email": "Sora322@hotmao.fr",
            "expoPushToken": "ExponentPushToken[5tFwMaOMWh4K4CTfpJzgRR]",
            "id": 17,
            "login": "Sora3",
            "name": "Dhdb",
            "password": "bf7420009c1334246948291c7df7925d25d8c1d845e822b54a418d28d6fc0a482c0229f06b177af26fb00fec0e876d13739e994af139f893cc61c7972929634c",
            "phone": null,
            "reward": 1,
            "rewardTier": 1,
            "rewardValue": 10,
            "tier": 1,
            "zipCode": null,
        }, {
            "additionalInformation": null,
            "address": null,
            "admin": 1,
            "avatar": "images/uploads/4c72d8ac63711879718a5b87055f41dd.png",
            "coins": 131,
            "country": null,
            "email": "charleslebastard@hotmail.fr",
            "expoPushToken": "",
            "id": 4,
            "login": "Nalmanak",
            "name": "Charles",
            "password": "bf7420009c1334246948291c7df7925d25d8c1d845e822b54a418d28d6fc0a482c0229f06b177af26fb00fec0e876d13739e994af139f893cc61c7972929634c",
            "phone": null,
            "reward": 1,
            "rewardTier": 1,
            "rewardValue": 10,
            "tier": 1,
            "zipCode": null,
        }, {
            "additionalInformation": null,
            "address": null,
            "admin": 0,
            "avatar": null,
            "coins": 110,
            "country": null,
            "email": "antoine.dury@epitech.eu",
            "expoPushToken": "",
            "id": 3,
            "login": "Anto",
            "name": "Antoine",
            "password": "bf7420009c1334246948291c7df7925d25d8c1d845e822b54a418d28d6fc0a482c0229f06b177af26fb00fec0e876d13739e994af139f893cc61c7972929634c",
            "phone": null,
            "reward": 1,
            "rewardTier": 1,
            "rewardValue": 10,
            "tier": 1,
            "zipCode": null,
        }, {
            "additionalInformation": null,
            "address": null,
            "admin": 0,
            "avatar": null,
            "coins": 110,
            "country": null,
            "email": "Momo@gmail.com",
            "expoPushToken": "",
            "id": 19,
            "login": "Momo",
            "name": "Momo",
            "password": "bf7420009c1334246948291c7df7925d25d8c1d845e822b54a418d28d6fc0a482c0229f06b177af26fb00fec0e876d13739e994af139f893cc61c7972929634c",
            "phone": null,
            "reward": 1,
            "rewardTier": 1,
            "rewardValue": 10,
            "tier": 1,
            "zipCode": null,
        }, {
            "additionalInformation": null,
            "address": null,
            "admin": 0,
            "avatar": "images/uploads/b77036a7cb9344dfc7b901e07097e036.png",
            "coins": 0,
            "country": null,
            "email": "a.a@a.fr",
            "expoPushToken": "",
            "id": 15,
            "login": "Altan",
            "name": "",
            "password": "43102984a36d9196dd2d92cf10de63cd44226bcc6f8e5135a9780a9c8303a44fb687ba6134b4d4b68259aac57d7ff0f2ab6ca877019b75752dda8cf03a1c18ac",
            "phone": null,
            "reward": 1,
            "rewardTier": 1,
            "rewardValue": 10,
            "tier": 1,
            "zipCode": null,
        }];
        resolve(res);
    }));
}

module.exports = UsersService;
