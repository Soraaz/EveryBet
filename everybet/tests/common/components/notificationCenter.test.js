import React from "react";
import renderer from "react-test-renderer";

import NotificationCenter from "../../../src/common/components/notificationCenter.js";
import {shallow} from "enzyme/build";
import {Notifications, Permissions} from 'expo';
import User from "../../../src/classes/user";
import UsersService from "../../../services/UsersService";
import AlertMessage from "../../../src/common/components/alertMessage";

describe("NotificationCenter", () => {
    describe("rendering", () => {
        it("Should render correctly", () => {
            const tree = renderer
                .create(<NotificationCenter/>)
                .toJSON();
            expect(tree).to.matchSnapshot();
        });

        describe("functionalities", () => {
            let component;
            let wrapper;
            let cb = sinon.spy();

            beforeEach(() => {
                component = new NotificationCenter();
                wrapper = shallow(<NotificationCenter callback={cb} />);
            });

            it("Should instantiate the class", () => {
                expect(component).to.be.an.instanceOf(NotificationCenter);
            });

            it("CheckToken", () => {
                component.askPermission = sinon.spy();
                expect(component.askPermission).to.be.not.called;
                component.checkToken();
                expect(component.askPermission).to.be.called;
            });

            it("Ask Permissions not granted for ios", (done) => {
                Permissions.getAsync = jest.fn().mockReturnValue(new Promise(async (resolve) => {
                    let data = {status: "not get"};
                    resolve(data);
                }));
                Permissions.askAsync = jest.fn().mockReturnValue(new Promise(async (resolve) => {
                    let data = {status: "not get"};
                    resolve(data);
                }));
                component.registerForPushNotificationsAsync = sinon.spy();
                expect(component.registerForPushNotificationsAsync).to.be.not.called;
                component.askPermission();
                setTimeout(() => {
                    expect(component.registerForPushNotificationsAsync).to.be.not.called;
                    done();
                }, 200);
            });

            it("Ask Permissions granted but acquired", (done) => {
                Permissions.getAsync = jest.fn().mockReturnValue(new Promise(async (resolve) => {
                    let data = {status: "not get"};
                    resolve(data);
                }));
                Permissions.askAsync = jest.fn().mockReturnValue(new Promise(async (resolve) => {
                    let data = {status: "granted"};
                    resolve(data);
                }));
                component.registerForPushNotificationsAsync = sinon.spy();
                expect(component.registerForPushNotificationsAsync).to.be.not.called;
                component.askPermission();
                setTimeout(() => {
                    expect(component.registerForPushNotificationsAsync).to.be.called;
                    done();
                }, 200);
            });

            it("Ask Permissions granted for ios", (done) => {
                Permissions.getAsync = jest.fn().mockReturnValue(new Promise(async (resolve) => {
                    let data = {status: "granted"};
                    resolve(data);
                }));
                component.registerForPushNotificationsAsync = sinon.spy();
                expect(component.registerForPushNotificationsAsync).to.be.not.called;
                component.askPermission();
                setTimeout(() => {
                    expect(component.registerForPushNotificationsAsync).to.be.called;
                    done();
                }, 200);
            });

            it("registerForPushNotificationsAsync already get token", (done) => {
                Notifications.addListener = jest.fn();
                Notifications.getExpoPushTokenAsync = jest.fn().mockReturnValue(new Promise(async (resolve) => {
                    let data = "token";
                    resolve(data);
                }));
                User.expoPushToken = "token";
                Notifications.addListener = sinon.spy();
                expect(Notifications.addListener).to.be.not.called;
                component.registerForPushNotificationsAsync();
                setTimeout(() => {
                    expect(Notifications.addListener).to.be.called;
                    done();
                }, 200);
            });

            it("registerForPushNotificationsAsync already new token", (done) => {
                Notifications.addListener = jest.fn();
                Notifications.getExpoPushTokenAsync = jest.fn().mockReturnValue(new Promise(async (resolve) => {
                    let data = "token";
                    resolve(data);
                }));
                User.expoPushToken = "another token";
                UsersService.updateUserById = jest.fn().mockReturnValue(new Promise(async (resolve) => {
                    let res = [
                        {
                            "test": "test",
                        },
                    ];
                    resolve(res);
                }));

                User.updateUser = sinon.spy();
                Notifications.addListener = sinon.spy();
                expect(Notifications.addListener).to.be.not.called;
                expect(User.updateUser).to.be.not.called;
                component.registerForPushNotificationsAsync();
                setTimeout(() => {
                    expect(Notifications.addListener).to.be.called;
                    expect(User.updateUser).to.be.called;
                    done();
                }, 200);
            });

            it("registerForPushNotificationsAsync already new token error", (done) => {
                Notifications.addListener = jest.fn();
                Notifications.getExpoPushTokenAsync = jest.fn().mockReturnValue(new Promise(async (resolve) => {
                    let data = "token";
                    resolve(data);
                }));
                User.expoPushToken = "another token";
                UsersService.updateUserById = jest.fn().mockReturnValue(new Promise(async (resolve) => {
                    let res = {'error' : 'Error: Test'};
                    resolve(res);
                }));

                AlertMessage.error = sinon.spy();
                Notifications.addListener = sinon.spy();
                expect(Notifications.addListener).to.be.not.called;
                expect(AlertMessage.error).to.be.not.called;
                component.registerForPushNotificationsAsync();
                setTimeout(() => {
                    expect(Notifications.addListener).to.be.called;
                    done();
                }, 200);
            });

            it("Handle notifications", (done) => {
                component.setState = sinon.spy();
                expect(component.setState).to.be.not.called;
                component._handleNotification("test");
                setTimeout(() => {
                    expect(component.setState).to.be.called;
                    done();
                }, 200);
            });

            it("Launch Crontab daily reward", (done) => {
                Notifications.scheduleLocalNotificationAsync = sinon.spy();
                expect(Notifications.scheduleLocalNotificationAsync).to.be.not.called;
                component._dailyRewardNotification();
                setTimeout(() => {
                    expect(Notifications.scheduleLocalNotificationAsync).to.be.called;
                    done();
                }, 200);
            });
        });
    });
});
