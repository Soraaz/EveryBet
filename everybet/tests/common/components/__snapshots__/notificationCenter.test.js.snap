// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationCenter rendering Should render correctly 1`] = `null`;
